### PR TITLE
feat(NativeScript) Add support for NativeScript

### PR DESCRIPTION
**Pull request by shiv19**
#### Description
These changes will allow TFJS to be used inside a NativeScript App

---
##### For repository owners only:

Please remember to apply all applicable tags to your pull request.
Tags: FEATURE, BREAKING, BUG, PERF, DEV, DOC, SECURITY

For more info see: https://github.com/tensorflow/tfjs/blob/master/DEVELOPMENT.md

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-core/1293)

### CODE REVIEW

**Comment by shiv19** [comment]
Here is the POC
https://github.com/shiv19/nativescript-tfjs-poc

**Comment by jlooper** [comment]
So exciting! Hope to see this sweet PR merged so we can tinker with TF.js!

File: src/util.ts [hunk 1]
```diff
@@ -508,6 +508,10 @@ export function now(): number {
   } else if (typeof process !== 'undefined') {
     const time = process.hrtime();
     return time[0] * 1000 + time[1] / 1000000;
+  } else if (global && global.android !== 'undefined') {
+    return java.lang.System.nanoTime() / 1000000;
+  } else if (global && global.ios !== 'undefined') {
```
Comment (NathanaelA) [inline src/util.ts]: There is no `global.ios` -- I've tried to get NS team to add one to make this simpler for us plugin developers -- but it doesn't exist; you would have to do something like `if (global && global.NSObject  && global.NSString) {` to determine if you are on iOS.

Comment (shiv19) [reply src/util.ts]: Ah! Let me update that. `global.android` is present, right?

**Comment by nsthorat** [comment]
It's great to see this, though I don't think we can maintain this in core since it's a little specific to a single execution environment which we don't have expertise in.

I have a suggestion, though, is this something we could do outside of the library? You can easily override `util.now()`, or `performance.now()` and patch in global as window.

**Comment by shiv19** [comment]
Hi @nsthorat 
Thanks for your suggestion,
like you said, I was able to get it working with changes made outside of the library.
Now I can just create a NativeScript plugin to do the same.
I'll close this PR now. Thanks for the idea :)

I wrote this code to get it working outside of Tfjs

```
global.setImmediate = setTimeout;
global.clearImmediate = clearTimeout;
global.performance = {
  now() {
    if (global.android) {
      return java.lang.System.nanoTime() / 1000000;
    } else {
      return CACurrentMediaTime();
    }
  }
};
global.process = {
  versions: {
    node: '10.0'
  }
};
```

It works fine, but there could be unintended consequences later. I'll update this thread if I find any such thing.

**Comment by nsthorat** [comment]
Great! Yes definitely let me know. If we try to understand what hooks we need for other things like react native we may be able to provide an API to set some of this stuff more explicitly.

**Pull request closed without merging**